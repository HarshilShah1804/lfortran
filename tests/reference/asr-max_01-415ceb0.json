{
    "basename": "asr-max_01-415ceb0",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/max_01.f90",
    "infile_hash": "3154045abbca74724d67eeb8e4744c66f702ec1a598db55701c43729",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-max_01-415ceb0.stdout",
    "stdout_hash": "eb9d7af4c76b60a99986df7442d82524e967f2de6fc02ad85dbfabbb",
    "stderr": "asr-max_01-415ceb0.stderr",
    "stderr_hash": "ffae0278fb457614873da2a97daf1953347eda5f0a0585016b6cf07c",
    "returncode": 0
}