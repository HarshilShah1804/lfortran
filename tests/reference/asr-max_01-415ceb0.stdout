(TranslationUnit
    (SymbolTable
        1
        {
            max_01:
                (Program
                    (SymbolTable
                        2
                        {
                            y:
                                (Variable
                                    2
                                    y
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z:
                                (Variable
                                    2
                                    z
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    max_01
                    []
                    [(Assignment
                        (Var 2 y)
                        (RealConstant
                            5.200000
                            (Real 8)
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 z)
                        (RealConstant
                            9.000000
                            (Real 4)
                        )
                        ()
                    )
                    (Print
                        (StringFormat
                            ()
                            [(IntrinsicElementalFunction
                                Max
                                [(Cast
                                    (Var 2 y)
                                    RealToReal
                                    (Real 8)
                                    ()
                                )
                                (Cast
                                    (Var 2 z)
                                    RealToReal
                                    (Real 8)
                                    ()
                                )]
                                0
                                (Real 8)
                                ()
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (If
                        (RealCompare
                            (IntrinsicElementalFunction
                                Abs
                                [(RealBinOp
                                    (IntrinsicElementalFunction
                                        Max
                                        [(Cast
                                            (Var 2 y)
                                            RealToReal
                                            (Real 8)
                                            ()
                                        )
                                        (Cast
                                            (Var 2 z)
                                            RealToReal
                                            (Real 8)
                                            ()
                                        )]
                                        0
                                        (Real 8)
                                        ()
                                    )
                                    Sub
                                    (RealConstant
                                        9.000000
                                        (Real 8)
                                    )
                                    (Real 8)
                                    ()
                                )]
                                0
                                (Real 8)
                                ()
                            )
                            Gt
                            (RealConstant
                                0.000000
                                (Real 8)
                            )
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            (StringConstant
                                "Incorrect value of max"
                                (String 1 22 () PointerString)
                            )
                        )]
                        []
                    )
                    (Print
                        (StringFormat
                            ()
                            [(StringConstant
                                "Kind of max:"
                                (String 1 12 () PointerString)
                            )
                            (TypeInquiry
                                Kind
                                (Real 8)
                                (IntrinsicElementalFunction
                                    Max
                                    [(Cast
                                        (Var 2 y)
                                        RealToReal
                                        (Real 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        RealToReal
                                        (Real 8)
                                        ()
                                    )]
                                    0
                                    (Real 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (If
                        (IntegerCompare
                            (TypeInquiry
                                Kind
                                (Real 8)
                                (IntrinsicElementalFunction
                                    Max
                                    [(Cast
                                        (Var 2 y)
                                        RealToReal
                                        (Real 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        RealToReal
                                        (Real 8)
                                        ()
                                    )]
                                    0
                                    (Real 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )
                            NotEq
                            (IntegerConstant 8 (Integer 4) Decimal)
                            (Logical 4)
                            (LogicalConstant
                                .false.
                                (Logical 4)
                            )
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
